### DISCUSSION TITLE
DS1302 two minutes delays after 7 days

### DISCUSSION THREAD

**Discussion by arshidkv12** [Makuna/Rtc#200] [Q&A] [answered]
IC: Ds1302 
Two minutes fast after 7 days. 
How to fix it?

**Comment by Makuna** [comment]
Need more details or explanation.  
What do you mean by "two minute delays"?  Do you mean you notice that the time returned from the RTC is incorrect and slow by two minutes over the period of 7 days?

The DS1302 is not a very time accurate RTC.  At best it loses several seconds a day, at worse it can lose several minutes per day.  It is very temperature sensitive.  Newer chips like the DS3231 include temperature compensation and are usually accurate to several seconds per month.

**Comment by arshidkv12** [comment]
Sorry... It's running 2 minutes ahead within the last 7 days. It's being powered by a 5V HiLink SMPS and a 3V cell.

**Comment by arshidkv12** [comment]
How to improve accuracy?

**Reply by Makuna** [reply]
The only way to improve the accuracy of the DS1302 is put it in a very temperature-controlled environment.  It still won't be very accurate.  Switch to using a another RTC that is more accurate (DS3231/DS3234/Pcf8563/BM8563).

**Reply by arshidkv12** [reply]
What is an affordable IC similar to the DS1302?
Decrease by 2 minutes after a 7-day period. What do you think?

**Reply by Makuna** [reply]
https://edadocs.software.keysight.com/kkbopen/timebase-inaccuracy-and-its-overall-effect-on-measurement-accuracy-588265262.html
DS3231/DS3232/DS3234
    Accuracy ±2ppm from 0°C to +40°C
    Accuracy ±3.5ppm from -40°C to +85°C
DS3231 is usually the next step up.

**Reply by arshidkv12** [reply]
Is the PCF8563/BM8563 more precise?

**Reply by Makuna** [reply, marked as answer]
it states 0.2ppm, but it doesn't have a temperature compensation that I know of.  I have never tested it myself though.
